src/main/java/tahub/contacts/storage/JsonAdaptedGrade.java

```diff
@@ -0,0 +1,60 @@
+package tahub.contacts.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import tahub.contacts.commons.exceptions.IllegalValueException;
+
+/**
+ * Jackson-friendly version of a grade entry.
+ */
+class JsonAdaptedGrade {
+
+    private final String assessmentName;
+    private final double score;
+    private final double weight;
+
+    /**
+     * Constructs a {@code JsonAdaptedGrade} with the given grade details.
+     */
+    @JsonCreator
+    public JsonAdaptedGrade(@JsonProperty("assessmentName") String assessmentName,
+                            @JsonProperty("score") double score,
+                            @JsonProperty("weight") double weight) {
+        this.assessmentName = assessmentName;
+        this.score = score;
+        this.weight = weight;
+    }
+
+    public String getAssessmentName() {
+        return assessmentName;
+    }
+
+    public double getScore() {
+        return score;
+    }
+
+    public double getWeight() {
+        return weight;
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted grade object into the model's grade data.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted grade.
+     */
+    public void toModelType(tahub.contacts.model.grade.GradingSystem gradingSystem) throws IllegalValueException {
```

Comment:
Hey thanks for the effort but this method **cannot** return void. Nor should it accept an input argument. It should return the non json version of whatever class this object is trying to represent. Correspondingly, toModelType_validGradeDetails_success() of JsonAdaptedGradeTest.java needs to be updated :)

Reply:
Hmmmm.right!!
Noted will change 
Thank you:)